base/src/closure.rs

```diff
@@ -0,0 +1,23 @@
+#![allow(clippy::type_complexity, clippy::too_many_arguments)]
+#[derive(Clone)]
+pub struct Closure0<Output: ?Sized>(core::marker::PhantomData<Output>);
+
+impl<Output: ?Sized> crate::TObj<Closure0<Output>> {
+    pub fn invoke(self) -> crate::TObj<Output> {
+        let f = self.obj.into_raw();
+        unsafe {
+            let res = crate::lean_apply_1(f, crate::lean_box(0));
+            crate::TObj::from_raw(res)
+        }
+    }
+}
+
+#[cfg(feature = "nightly")]
+impl<Output: ?Sized> FnOnce<()> for crate::TObj<Closure0<Output>> {
+    type Output = crate::TObj<Output>;
+    extern "rust-call" fn call_once(self, _args: ()) -> Self::Output {
+        self.invoke()
+    }
+}
+
+include!(concat!(env!("OUT_DIR"), "/fixed_closures.rs"));
```

Comment:
I think you could do this with a macro-rules macro

Reply:
It seems hard to manipulate those argument names. Could you give me some insights?

Reply:
I'll push an implementation shortly. Name generation is the one part you can't really do in macro_rules, so you end up with a list of names in the macro call, but otherwise it can all be done declaratively.